halo2_proofs/src/circuit/value.rs

```diff
@@ -612,6 +612,17 @@ impl<F: Field> Mul<F> for Value<&Assigned<F>> {
     }
 }
 
+impl<V, E> Value<Result<V, E>> {
+    /// Transposes an `Value` of a [`Result`] into a [`Result`] of an `Value`.
+    pub fn transpose(self) -> Result<Value<V>, E> {
+        match self.inner {
+            Some(Ok(x)) => Ok(Value::known(x)),
+            Some(Err(e)) => Err(e),
+            None => Ok(Value::unknown()),
+        }
+    }
+}
+
```

Comment:
Sorry for nitpicking, I think this will have the same issue with the impl of `PartialCmp`, since it's output is not just a `Value` and it depends on its inner value.

It's fine to keep this API for now just for EVM circuit usage, since I believe most circuits won't use this, but I still prefer to refactor it out in the future.

Reply:
That sounds good to me, thanks again!